### ISSUE TITLE
parse(PhoneNumberUtil.normalizeDigitsOnly(number), "ZZ") throws NumberParseException

### ISSUE THREAD

**Issue by mrhota** [googlei18n/libphonenumber#763]
My goal is to correctly parse "+610412341234" but fail to parse "+6104foobarby" using the two calls above.

`normalizeDigitsOnly(number)` strips the leading "+" from an international number, so I can't give it directly to the `parse(number, "ZZ")` method. It seems like a waste to have to prepend a `+` which was already there to begin with. Providing an alphanumeric phone number to `parse(number, "ZZ")` converts the alphabetic characters to digits, which is explicitly _not_ desired.

I could of course do a bit of string manipulation before running parse, but then what's the point of `normalizeDigitsOnly` (I can remove non-numbers with `replaceAll` and `\D`)?

Is this possible? Or perhaps a new normalization method (like `normalizeDigitsOnly`) could leave a leading `+`?

**Comment by keghani** [comment]
The javadoc at https://github.com/googlei18n/libphonenumber/tree/master/java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java for normalizedigitsonly(String): "strips punctuation and alpha characters".
It's working as intended.

Similarly with parse(String, String), it's meant to parse alphabetic characters.

As to the point of normalizeDigitsOnly - it "converts wide-ascii and arabic-indic numerals to European numerals" (from the javadoc), i.e. normalizes them.

I think what you're looking for is something that calls normalizeDigits(String number, boolean keepNonDigits) with true. Do I understand you correctly?

**Comment by mrhota** [comment]
I figured `normalizeDigitsOnly` _might_ leave the leading `+`, as it's important, and, in the case of `parse`, it's required if the locale is "ZZ". But I understand why it doesn't. I also understand why `parse` will convert alphas to numerals.

What I'm looking for is a _simple_ and straightforward way to achieve (using libphonenumber functionality) the following: parse `+610412341234` and reject `+6104foobarba`.

It seemed like something that might have come up before. That's why I suggested providing an additional normalization method, one which leaves leading '+' and strips all other non-digits. It would behave _exactly the same_ as normalizeDigitsOnly, but would leave the initial '+'.

A slight clarification on the method signature you provided: `normalizeDigits(String number, boolean keepLeadingPlus)`. Or perhaps without a boolean param (because `false` would be the same as `normalizeDigitsOnly`): `normalizeDigitsKeepLeadingPlus(String number)`
